shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -243,6 +243,12 @@ object Utility extends AnyRef with parsing.TokenTests {
       }
     }
 
+  // Checks if node when converted to string is an entity ref
+  def checkNodeForEntityRef(n: Node): Boolean = {
+    val st = n.toString
+    st(0) == '&' && st(st.length - 1) == ';'
```

Comment:
I'm not sure if we can completely get rid of `checkNodeForEntityRef`, as a test seems to fail when using 

`if(!prev.isAtom && !x.isAtom) sb.append(' ')`
instead of
`if (!checkNodeForEntityRef(prev) && !checkNodeForEntityRef(x)) sb.append(' ')`

![1a7268b4-a67b-4ae7-8255-edf3af10abfb](https://cloud.githubusercontent.com/assets/17111672/24908549/08abc780-1ede-11e7-8f4c-73a36ce17ded.jpg)

I'm struggling at finding the alternatives to this definition.